utils/rpms/daos.spec

```diff
@@ -158,7 +158,11 @@ Requires: openssl
 # This should only be temporary until we can get a stable upstream release
 # of mercury, at which time the autoprov shared library version should
 # suffice
+%if 0%{ucx} > 0
+Requires: mercury-ucx >= %{mercury_version}
+%else
 Requires: mercury >= %{mercury_version}
+%endif
```

Comment:
Just to note that while this may be useful for testing this cannot land.  daos cannot ever `Requires: mercury-ucx`.  It can only `Requires: mercury`.  The choice of using [*mercury-*]*ucx* is made by the end user by them installing `mercury-ucx` themselves if they want UCX.

So in a final iteration of a PR that is intended to land and enable UCX on hardware nodes in our test pipeline, the post-provisioning daos install will have to handle the installation of `mercury-ucx` where UCX is to be used.

Reply:
This PR is for testing and not intended to land.  The installation of `mecury-ucx` is being implemented differently in https://github.com/daos-stack/daos/pull/8902

Reply:
Fair enough.  I figured you knew that but just wanted to be explicit about it (for anyone else reading who might not be aware) as there is the potential for confusion about how this whole optional UCX mechanism is supposed to work.